main.go

```diff
@@ -182,7 +184,16 @@ func generateCharts(c *cli.Context) {
 			// New chart was introduced
 			modifiedAssetPath = fmt.Sprintf("%s/*", filepath.Join(path.RepositoryAssetsDir, splitPath[1], splitPath[2]))
 		default:
-			// Existing chart was modified
+			if strings.HasSuffix(splitPath[2], "-crd") {
+				// e.g. p = charts/rancher-monitoring/rancher-monitoring-crd/100.1.0+up19.0.3/files/crd-manifest.tgz
+				chartKey := fmt.Sprintf("%s-%s-%s", splitPath[1], splitPath[2], splitPath[3])
+				if strings.HasSuffix(p, ".tgz") {
+					modifiedCrdTgzPaths[chartKey] = p
+					continue
+				}
+				modifiedCrdNormalFiles[chartKey] = true
+			}
```

Comment:
I think we should just skip if the CRD tgz file changes, since we know the tgz file will always change... And if an underlying file was modified, it would have added the CRD chart to the loop anyways.

Why can't we just modify all this logic to:

```go
if strings.HasSuffix(p, ".tgz") {
	// e.g. p = charts/rancher-monitoring/rancher-monitoring-crd/100.1.0+up19.0.3/files/crd-manifest.tgz
	continue
}
```

Reply:
Talked to Jack offline; the issue here is that we need to add the underlying `crd-manifest.tgz` into the `wt.Excludes`, and we only want to add it once. This is why an intermediary map is required,